File: CodeForces/201-300/271/A/271A.py
# Problem 271A: Beautiful Year
# https://codeforces.com/contest/271/problem/A
# Executed with: Python 3.11.2
# Linux version: Linux kernel 6.1.0-10-amd64
# Submitted on: September 26th, 2024
# Codeforces language used: PyPy 3.10 (7.3.15, 64bit)

year = int(input())

while True:
    year += 1
    nums = {int(digit) for digit in str(year)}
    if len(nums) == 4:
        print(year)
        break

File: CodeForces/201-300/271/A/271A.ts
// Problem 271A: Beautiful Year
// https://codeforces.com/contest/271/problem/A
// Compiled with: npx 9.2.0, tsc Version 5.9.3
// Executed with: node v20.19.4
// Linux version: Linux kernel 6.17.0-5-generic
// Written on: October 26th, 2025
// Codeforces language used: Cannot be submitted

import * as readline from 'node:readline/promises';
import { stdin as input, stdout as output } from 'node:process';

async function main() {
    const rl = readline.createInterface({ input, output });

    const line: string = await rl.question('');
    let year: number = parseInt(line.trim(), 10);
    rl.close();

    while (true) {
        year += 1;
        let arr: string[] = year.toString().split('');
        let numSet = new Set<string>(arr);
        if (numSet.size == 4) {
            console.log(year);
            return;
        }
    }

}

main();
